wallet/src/extkey.rs

```diff
@@ -205,31 +212,21 @@ impl ExtendedKey {
 	/// Return the identifier of the key, which is the
 	/// Hash160 of the private key
 	pub fn identifier(&self) -> Identifier {
-		let mut sha = Sha256::new();
-		sha.input(&self.key[..]);
-
-		let mut shres = [0; 32];
-		sha.result(&mut shres);
-
-		let mut ripe = Ripemd160::new();
-		ripe.input(&shres[..]);
-
-		let mut identifier = [0; 20];
-		ripe.result(&mut identifier);
+		let digest = sha256::hash(&self.key[..]);
+		let identifier = Ripemd160::digest(&digest[..]);
 		Identifier::from_bytes(&identifier)
 	}
 
 	/// Derive an extended key from an extended key
 	pub fn derive(&self, secp: &Secp256k1, n: u32) -> Result<ExtendedKey, Error> {
-		let mut hmac = Hmac::new(Sha512::new(), &self.chaincode[..]);
+		let hmac = hmacsha512::Key(self.chaincode.clone());
+
 		let mut n_bytes: [u8; 4] = [0; 4];
 		BigEndian::write_u32(&mut n_bytes, n);
+		let mut seed = self.key[..].to_vec();
+		seed.extend_from_slice(&n_bytes);
 
-		hmac.input(&self.key[..]);
-		hmac.input(&n_bytes[..]);
-
-		let mut derived = [0; 64];
-		hmac.raw_result(&mut derived);
+		let derived = hmacsha512::authenticate(&seed, &hmac);
```

Comment:
This is a good example of how sodiumoxide/libsodium api is more user friendly and less prone to error.